sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1063,7 +1063,7 @@ object functions extends LegacyFunctions {
    *
    * @group normal_funcs
    */
-  def expr(expr: String): Column = Column(SqlParser.parseExpression(expr))
+  def expr(expr: String): Column = Column(CatalystQl.parseExpression(expr))
```

Comment:
Not pretty but: you could get the conf from the SQLContext getOrCreate / getActive?